### DISCUSSION TITLE
CockroachDB, PostgreSQL backend plugin for Zenoh

### DISCUSSION THREAD

**Discussion by clibequilibrium** [eclipse-zenoh/roadmap#66] [Ideas and new features]
Hello Zenoh community !

I would like to start discussion and propose an idea of using SQL Backend as a storage for Zenoh.
CockroachDB is gaining a huge popularity among distributed cloud databases. From Cockroach FAQ we can see that

> What is CockroachDB used for?
CockroachDB is designed to survive software and hardware failures, from server restarts to datacenter outages. This is accomplished without confusing artifacts typical of other distributed systems (e.g., stale reads) using strongly-consistent replication as well as automated repair after failures

And do we see any benefit in it ? I think it will give Zenoh the ability to cover many use cases, from gaming, to enterprise once Zenoh gets an SQL backend. Currently the choices of Zenoh backends are, fs, RocksDB (which CockroachDB is based on), influxdb 1.8 , and S3 (in preview) and it seems that SQL backend is what missing to make Zenoh more appealing for cloud ! Currently  we are relying on eventual consistency replication of Zenoh, or S3 backend. influxdb on other hand offers that only in Enterprise version of the database; while CockroachDB was made with high availability and high resilience in mind.

What are your thoughts on that? I will be happy to discuss !

**Comment by kydos** [comment]
The SQL backend is on our todo-list, CockroachDB seems like a good candidate. Would you be interested in contributing the plugin?

**Reply by clibequilibrium** [reply]
> The SQL backend is on our todo-list, CockroachDB seems like a good candidate. Would you be interested in contributing the plugin? 

I would be happy to contribute if I could. I use Zenoh-C bindings and not sure if will be able to assemble a plugin in Rust.

**Reply by kydos** [reply]
We are going through the feature list for the next release. We'll try to squeeze this in. Do you have any specific timelines? 
We can do the plugin, but it would be great if you could help with testing in preparation for the release.

**Reply by clibequilibrium** [reply]
> We are going through the feature list for the next release. We'll try to squeeze this in. Do you have any specific timelines? 
> 
> We can do the plugin, but it would be great if you could help with testing in preparation for the release.

I don't think it's a priority for me. Let's ask Discord community if anyone is in need of sql backend? I am currently using S3
